frappe/auth.py

```diff
@@ -141,23 +141,26 @@ def __init__(self):
 
 	def login(self):
 		# clear cache
-		frappe.clear_cache(user=frappe.form_dict.get("usr"))
-		user, pwd = get_cached_user_pass()
-		self.authenticate(user=user, pwd=pwd)
-		if self.force_user_to_reset_password():
-			doc = frappe.get_doc("User", self.user)
-			frappe.local.response["redirect_to"] = doc.reset_password(
-				send_email=False, password_expired=True
-			)
-			frappe.local.response["message"] = "Password Reset"
-			return False
-
-		if should_run_2fa(self.user):
-			authenticate_for_2factor(self.user)
-			if not confirm_otp_token(self):
+		if frappe.get_system_settings("disable_user_pass_login"):
```

Comment:
@ankush - I am not sure if this check here will block all API requests for login or not (via direct POST request or `/api/method`)